src/super_gradients/training/datasets/detection_datasets/detection_dataset.py

```diff
@@ -468,11 +508,24 @@ def apply_transforms(self, sample: Dict[str, Union[np.ndarray, Any]]) -> Dict[st
         :param sample: Sample to apply the transforms on to (loaded with self.get_sample)
         :return: Transformed sample
         """
+
+        has_crowd_target = "crowd_target" in sample
+        detection_sample = self.__convert_input_dict_to_detection_sample(sample)
+        target_format_transform: Optional[DetectionTargetsFormatTransform] = None
+
         for transform in self.transforms:
-            sample["additional_samples"] = self._get_additional_inputs_for_transform(transform=transform)
-            sample = transform(sample=sample)
-            sample.pop("additional_samples")  # additional_samples is not useful after the transform
-        return sample
+            detection_sample.additional_samples = [
+                self.__convert_input_dict_to_detection_sample(s) for s in self._get_additional_inputs_for_transform(transform=transform)
+            ]
+            detection_sample = transform.apply_to_sample(sample=detection_sample)
+            detection_sample.additional_samples = None
+            if isinstance(transform, DetectionTargetsFormatTransform):
```

Comment:
This is a bit sketchy here...
So if some user for some reason puts target transform in the middle of the pipeline and we just apply it at the end behind the scenes?
Was this intended?

Reply:
It is sketchy, that is for sure. However this DetectionTargetsFormatTransform anyway should go last (It even listed in documentation). So if user mess up - it will be messed up anyway. 
I honestly see no other way in doing this.